addons/utm/models/utm_source.py

```diff
@@ -2,11 +2,91 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 
-from odoo import fields, models
+from odoo import _, api, fields, models, tools
 
 
 class UtmSource(models.Model):
     _name = 'utm.source'
     _description = 'UTM Source'
 
-    name = fields.Char(string='Source Name', required=True, translate=True)
+    name = fields.Char(string='Source Name', required=True)
+
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_names = self.env['utm.mixin']._get_unique_names(self._name, [vals.get('name') for vals in vals_list])
+        for vals, new_name in zip(vals_list, new_names):
+            vals['name'] = new_name
+        return super(UtmSource, self).create(vals_list)
+
+    def _generate_name(self, record, content):
+        """Generate the UTM source name based on the content of the source."""
+        if not content:
+            return False
+
+        content = content.replace('\n', ' ')
+        if len(content) >= 24:
+            content = f'{content[:20]}...'
+
+        create_date = record.create_date or fields.date.today()
+        create_date = fields.date.strftime(create_date, tools.DEFAULT_SERVER_DATE_FORMAT)
+        model_description = self.env['ir.model']._get(record._name).name
+        return _('%s (%s created on %s)', content, model_description, create_date)
```

Comment:
Maybe use named variables to ease translations ? Like ``_('%(content)s (%(model_description)s created on %(create_date)s') ...`` ?

Reply:
good idea :)